xarray/core/alignment.py

```diff
@@ -478,6 +491,20 @@ def assert_unindexed_dim_sizes_equal(self) -> None:
                     f"because of conflicting dimension sizes: {sizes!r}" + add_err_msg
                 )
 
+    def assert_equal_dimension_names(self) -> None:
+        # When broadcasting is disabled, only allows objects having the exact same dimensions' names.
+        if self.broadcast:
+            return
+
+        unique_dims = set(tuple(o.sizes) for o in self.objects)
```

Comment:
```suggestion
        unique_dims = set(tuple(o.dims) for o in self.objects)
```

This seems a little clearer